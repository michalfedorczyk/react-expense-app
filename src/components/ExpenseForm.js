import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";
import "react-dates/lib/css/_datepicker.css";

const now = moment();
console.log(now.format("MMM Do, YYYY"));

export default class ExpenseForm extends React.Component {
  state = {
    description: "",
    note: "",
    amount: "",
  };

  onDescriptionChange = (e) => {
    this.setState(() => ({ description: e.target.value }));
  };

  onNoteChange = (e) => {
    this.setState(() => ({ note: e.target.value }));
  };

  onAmountChange = (e) => {
    const amount = e.target.value;
    if (amount.match(/^\d*(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  render() {
    return (
      <div>
        <form>
          <input
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            value={this.state.amount}
            type="text"
            placeholder="amount"
            onChange={this.onAmountChange}
          />
          <textarea
            placeholder="Add a note for your expense (optional)"
            value={this.state.note}
            onChange={this.onNoteChange}
          ></textarea>
          <button>Add Expense</button>
        </form>
        <h3>Expense Form</h3>
      </div>
    );
  }
}
