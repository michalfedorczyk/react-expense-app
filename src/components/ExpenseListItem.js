import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, description, note, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>Description: {description}</h3>
    </Link>
    <p>Note: {note}</p>
    <p>
      Amount: {amount} - created at: {createdAt}
    </p>
  </div>
);

export default ExpenseListItem;
