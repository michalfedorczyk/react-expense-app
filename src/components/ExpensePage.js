import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";

const ExpensePage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpensePage;
