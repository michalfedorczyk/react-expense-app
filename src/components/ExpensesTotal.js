import React from "react";
import { connect } from "react-redux";
import selectTotalExpenses from "../selectors/expenses-total";
import selectExpenses from "../selectors/expenses";
import numeral from "numeral";

export const ExpensesTotal = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formattedExpensesTotal = numeral(expensesTotal).format("$0,0.00");
  return (
    <div>
      <h4>
        Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}.
      </h4>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectTotalExpenses(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesTotal);
