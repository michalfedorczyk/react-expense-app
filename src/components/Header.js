import React from "react";

const Header = (props) => (
  <div>
    <h1>{props.title}</h1>
    <p>{props.subtitle}</p>
  </div>
);

Header.defaultProps = {
  title: "Indecision App",
};

export default Header;
