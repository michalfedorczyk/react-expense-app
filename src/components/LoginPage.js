import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">
        <i class="fas fa-money-bill-alt"> </i> Expenses
      </h1>
      <p>Start to manage your expenses.</p>
      <button className="button" onClick={startLogin}>
        <i className="fab fa-google"></i> Login with Google
      </button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
