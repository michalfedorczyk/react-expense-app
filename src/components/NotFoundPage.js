import React from "react";
import { Link } from "react-router-dom";

const NotFoundPage = () => (
  <div>
    <h1>
      404! - <Link to="/dashboard">Go to Homepage</Link>
    </h1>
  </div>
);

export default NotFoundPage;
