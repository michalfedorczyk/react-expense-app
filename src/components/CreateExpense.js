import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/expenses";

const CreateExpense = (props) => (
  <div>
    <h1>Create Expense</h1>
    <ExpenseForm
      onSubmit={(expense) => {
        props.dispatch(addExpense(expense));
        props.history.push("/");
      }}
    />
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: state.expenses,
  };
};

export default connect(mapStateToProps)(CreateExpense);
