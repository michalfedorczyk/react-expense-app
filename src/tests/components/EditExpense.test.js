import React from "react";
import { shallow } from "enzyme";
import { CreateExpense } from "../../components/CreateExpense";
import expenses from "../fixtures/expenses";

test("should ", () => {
  true;
});
