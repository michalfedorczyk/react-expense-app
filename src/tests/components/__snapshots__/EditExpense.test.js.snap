// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the edit expense correctly 1`] = `
<div>
  <h1>
    Edit Expense
  </h1>
  <ExpenseForm
    expense={
      Object {
        "amount": 1,
        "createdAt": 10368000000,
        "description": "Jamnik",
        "id": "3",
        "note": "Pies Jamnik",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove expense
  </button>
</div>
`;
